tests/integration/test_logcollector/test_configuration/test_basic_configuration_only_future_events.py

```diff
@@ -0,0 +1,105 @@
+
+]
+
+metadata = [
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': 'yes', 'valid_value': True},
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': 'no', 'valid_value': True},
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': 'yesTesting', 'valid_value': False},
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': 'noTesting', 'valid_value': False},
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': 'testingvalue', 'valid_value': False},
+    {'location': 'Security', 'log_format': 'eventchannel', 'only-future-events': '1234', 'valid_value': False}
+
+]
+
+configurations = load_wazuh_configurations(configurations_path, __name__,
+                                           params=parameters,
+                                           metadata=metadata)
+configuration_ids = [f"{x['LOCATION'], x['LOG_FORMAT'], x['ONLY_FUTURE_EVENTS']}" for x in parameters]
+
+
+def check_only_future_events_valid(cfg):
+    """Check if the Wazuh run correctly with the specified only future events field.
+
+    Ensure logcollector allows the specified future events attribute.
+
+    Raises:
+        TimeoutError: If the "Analyzing file" callback is not generated.
+    """
+    log_callback = logcollector.callback_eventchannel_analyzing(cfg['location'])
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=logcollector.GENERIC_CALLBACK_ERROR_ANALYZING_EVENTCHANNEL)
+
+def check_only_future_events_invalid(cfg):
+    """Check if the Wazuh fails because a invalid only future events configuration value.
+
+    Args:
+        cfg (dict): Dictionary with the localfile configuration.
+
+    Raises:
+        TimeoutError: If error callbacks are not generated.
+    """
+    log_callback = gc.callback_invalid_value('only-future-events', cfg['only-future-events'],
+                                             prefix, severity="WARNING")
+    wazuh_log_monitor.start(timeout=5, callback=log_callback,
+                            error_message=gc.GENERIC_CALLBACK_ERROR_MESSAGE)
+
+# fixtures
+@pytest.fixture(scope="module", params=configurations, ids=configuration_ids)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+def test_only_future_events(get_configuration, configure_environment, restart_logcollector):
+    """Check if the Wazuh only future events field of logcollector works properly.
+
+    Ensure Wazuh component fails in case of invalid values and works properly in case of valid
+    only future events values.
+
+    Raises:
+        TimeoutError: If expected callbacks are not generated.
+    """
+    cfg = get_configuration['metadata']
+    if cfg['valid_value']:
+        check_only_future_events_valid(cfg)
+    else:
+        check_only_future_events_invalid(cfg)
+
+
```

Comment:
```suggestion
    """Check if Wazuh only future events field of logcollector works properly.

    Ensure Wazuh component fails in case of invalid values and works properly in case of valid
    only future events values.

    Raises:
        TimeoutError: If expected callbacks are not generated.
    """
    cfg = get_configuration['metadata']
    if cfg['valid_value']:
        check_only_future_events_valid(cfg)
    else:
        check_only_future_events_invalid(cfg)
```